core/services/workflows/engine.go

```diff
@@ -191,26 +202,66 @@ func (e *Engine) Close() error {
 	})
 }
 
-func NewEngine(lggr logger.Logger, registry types.CapabilitiesRegistry) (*Engine, error) {
-	ctx := context.Background()
-	trigger, err := registry.GetTrigger(ctx, "on_mercury_report")
+func NewEngine(lggr logger.Logger, registry types.CapabilitiesRegistry) (engine *Engine, err error) {
+	engine = &Engine{
+		logger:     lggr.Named("WorkflowEngine"),
+		registry:   registry,
+		callbackCh: make(chan capabilities.CapabilityResponse),
+	}
+
+	// Trigger
+	engine.triggerType = "on_mercury_report"
+	engine.triggerConfig, err = values.NewMap(
+		map[string]any{
+			"feedlist": []any{
+				123, 456, 789, // ETHUSD, LINKUSD, USDBTC
+			},
+		},
+	)
 	if err != nil {
 		return nil, err
 	}
-	consensus, err := registry.GetConsensus(ctx, "off-chain-reporting")
+
+	// Consensus
+	engine.consensusType = "offchain_reporting"
+	engine.consensusConfig, err = values.NewMap(map[string]any{
+		"aggregation_method": "data_feeds_2_0",
+		"aggregation_config": map[string]any{
+			// ETHUSD
+			"0x1111111111111111111100000000000000000000000000000000000000000000": map[string]any{
+				"deviation": decimal.NewFromFloat(0.003),
+				"heartbeat": 24,
+			},
+			// LINKUSD
+			"0x2222222222222222222200000000000000000000000000000000000000000000": map[string]any{
+				"deviation": decimal.NewFromFloat(0.001),
+				"heartbeat": 24,
+			},
+			// BTCUSD
+			"0x3333333333333333333300000000000000000000000000000000000000000000": map[string]any{
+				"deviation": decimal.NewFromFloat(0.002),
+				"heartbeat": 6,
+			},
+		},
+		"encoder": "EVM",
+		"encoder_config": map[string]any{
+			"abi": "mercury_reports bytes[]",
+		},
+	})
 	if err != nil {
 		return nil, err
 	}
-	target, err := registry.GetTarget(ctx, "write_polygon_mainnet")
+
+	// Target
+	engine.targetType = "write_polygon-testnet-mumbai"
```

Comment:
You caught the casing change here because of selectors 👍🏻 In the future I was thinking about replacing `-` -> `_` to make the string more uniform, wdyt?

Reply:
SGTM!